common/src/main/java/org/opentestsystem/rdw/reporting/common/sqlbuilder/QueryProvider.java

```
@@ -66,6 +67,24 @@ public String newQueryAsString(final String templateName, final Iterable<String>
         return newQuery(templateName, addons).toString();
     }
 
+    private boolean addFrom(final SelectQuery query, final SqlTemplate template, final Iterable<String> addons) {
+        boolean containsFrom = template.getClauses().containsKey(FROM);
+        if (containsFrom) {
+            query.addCustomFromTable(template.getClauses().get(FROM));
+            return true;
+        }
+
+        for (final String addon : addons) {
+            checkArgument(template.getAddons() != null && template.getAddons().containsKey(addon), "missing sqlbuilder addon template");
```

This seems a bit inconsistent: this will throw *and* there is a meaningful return value. I'd change this method to throw instead of returning false. And perhaps call it `checkFrom` to imply that it will fly through or throw.

---

Ok, changed.